src/lib/sfge/SfgeWrapper.ts

```diff
@@ -164,27 +150,89 @@ export class SfgeWrapper extends CommandLineSupport {
 		if (this.ignoreParseErrors != null) {
 			args.push(`-DSFGE_IGNORE_PARSE_ERRORS=${this.ignoreParseErrors.toString()}`);
 		}
-		args.push(MAIN_CLASS, this.command, inputFile);
+		args.push(MAIN_CLASS, this.command, ...(await this.getSupplementalArgs()));
 
 		this.logger.trace(`Preparing to execute sfge with command: "${command}", args: "${JSON.stringify(args)}"`);
 		return [command, args];
 	}
 
-	private async execute(): Promise<string> {
+	protected abstract getSupplementalArgs(): Promise<string[]>;
+
+	protected async execute(): Promise<string> {
 		return super.runCommand();
 	}
+}
 
-	public static async getCatalog() {
-		const wrapper = await SfgeWrapper.create({
-			targets: [],
-			projectDirs: [],
+export class SfgeCatalogWrapper extends SfgeWrapper {
```

Comment:
Separated the wrapper into catalog wrapper and execution wrapper, analogous to PMD. Most of the behavior is the same, though.